gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/DefaultPubSubRpc.java

```diff
@@ -139,10 +140,20 @@ public DefaultPubSubRpc(PubSubOptions options) throws IOException {
         subBuilder.provideChannelWith(channel, true);
       } else {
         GoogleCredentials credentials = options.authCredentials().credentials();
-        pubBuilder.provideChannelWith(
-            credentials.createScoped(PublisherSettings.DEFAULT_SERVICE_SCOPES));
-        subBuilder.provideChannelWith(
-            credentials.createScoped(SubscriberSettings.DEFAULT_SERVICE_SCOPES));
+        ConnectionSettings pubConnectionSettings = ConnectionSettings.newBuilder()
+            .setServiceAddress(options.host())
+            .setPort(PublisherSettings.DEFAULT_SERVICE_PORT)
+            .provideCredentialsWith(
+                credentials.createScoped(PublisherSettings.DEFAULT_SERVICE_SCOPES))
+            .build();
+        ConnectionSettings subConnectionSettings = ConnectionSettings.newBuilder()
```

Comment:
It would make the code more readable to use a helper here. The initialization code for pub and sub are mostly identical.